test/e2e/app-dir/app/index.test.ts

```diff
@@ -317,16 +317,16 @@ createNextDescribe(
         try {
           // Click the link.
           await browser.elementById('link-to-rewritten-path').click()
-          await browser.waitForElementByCss('#from-dashboard')
+          await browser.waitForElementByCss('#from-dashboard', 5000)
```

Comment:
Why the 5000ms limit? Using an arbitrary value here can cause flakiness in tests similar to `waitFor(5000)`

Reply:
@ijjk I wasn't sure why these two are failing and I _guess_ it was flakiness caused by my change, but nope :(

So both tests are passing locally with both `next start` and `next dev`, and so far I have no luck why they're timing out here...

Reply:
I'm seeing the timeout locally as well, it looks like the issue is when it first clicks on the link the page is doing a full reload to the same URL instead of the URL of the link so then the test times out waiting for the new content but it doesn't actually navigate unless you click the link again after the full reload

Potentially this is from compilation taking longer as adding a `await next.render('/dashboard')` before the `await next.browser()` resolves this issue.